### ISSUE TITLE
Linux package bot failing - pub crash in observatory build

### ISSUE THREAD

**Issue by whesse** [dart-lang/sdk#23846]
With the new version of pub, the linux src package bot is failing.  There is a crash in pub during the observatory build:

https://uberchromegw.corp.google.com/i/client.dart/builders/linux-distribution-support-debian_wheezy-be/builds/7168/steps/Create%20src%20tarball/logs/stdio

```
Uncaught Error: Git error. Command: git rev-parse HEAD
fatal: Not a git repository (or any of the parent directories): .git
Stack Trace:
#0      runSync (package:pub/src/git.dart:75:24)
#1      PubCommandRunner._checkDepsSynced (package:pub/src/command_runner.dart:171:21)
#2      PubCommandRunner.runCommand.<runCommand_async_body> (package:pub/src/command_runner.dart:94:5)
#3      Future.Future.microtask.<anonymous closure> (dart:async/future.dart:144:37)
#4      _microtaskLoop (dart:async/schedule_microtask.dart:43:11)
#5      _microtaskLoopEntry (dart:async/schedule_microtask.dart:52:5)
#6      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96:13)
#7      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149:5)

#0      _rootHandleUncaughtError.<anonymous closure> (dart:async/zone.dart:895:5)
#1      _microtaskLoop (dart:async/schedule_microtask.dart:43:11)
#2      _microtaskLoopEntry (dart:async/schedule_microtask.dart:52:5)
#3      _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96:13)
#4      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:149:5)
make[2]: *** [runtime/observatory/pubspec.lock] Error 255
```

@nex3 @johnmccutchan

**Comment by nex3** [comment]
This looks like the code I added to make sure pub is running from the correct revision. Is there anything unusual about the build setup for this bot? Is pub not checked out from git, somehow?

**Comment by whesse** [comment]
That is correct, the debian packages are built from the source tarball, and
the whole point of a source tarball is that the package is compiled from
the files contained in the tarball, with no additional checkouts.  You have
to revert this check.  If you want, you could have a check that _if_ pub is
checked out from git, then it is the correct revision.

We cannot release any new versions of Dart until this is reverted.

On Wed, Jul 15, 2015 at 9:28 PM, Natalie Weizenbaum <
notifications@github.com> wrote:

> This looks like the code I added to make sure pub is running from the
> correct revision. Is there anything unusual about the build setup for this
> bot? Is pub not checked out from git, somehow?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dart-lang/sdk/issues/23846#issuecomment-121721133>.
>



-- 
William Hesse

**Comment by dgrove** [comment]
I think we should revert the change so that a new release can get built.

**Comment by nex3** [comment]
I have https://codereview.chromium.org/1246713005 out to fix this.

**Comment by turnidge** [comment]
Natalie - is this bug still alive?

**Comment by nex3** [comment]
No, it was fixed in dart-lang/pub@8abe19766853ad58c4c107a73678e2bccfeb1d4b.
